### PR TITLE
Added base README and example README template

### LINKED ISSUE #5: Top index
Please create the root level index ReadMe.md pointing and describing all the examples (I believe there will be more of them with time)
That will help a lot for a stranger approaching this repo at the first time

### PR DESCRIPTION
**Pull request by nickwesselman**
Anything else we should add to these @anastasiya29?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Sitecore Headless Samples and Examples
+This repository provides Sitecore and community-developed examples of various implementation and integration scenarios, for implementations which utilize Sitecore Headless SDKs. While Sitecore already provides official boilerplate and starters for these frameworks, questions about how to do X or Y often come up in community channels. The aim for this repository is to build a library of samples to help address these questions and provide a reference for Sitecore developers.
```
Comment (anastasiya29) [inline README.md]: Sounds a little wordy with "implementation" repeated. Also we're staying away from "starters", right? (Friends don't let friends use boilerplate as a starter) How about...

```suggestion
This repository provides various examples of Sitecore Headless SDKs implementations provided by Sitecore and community developers. While Sitecore already provides official boilerplate and samples for the different frameworks supported by the Headless SDKs, sometimes questions about use cases not covered by these samples pop up multiple times in community channels. The aim for this repository is to build a library of samples to help address these common questions and provide a reference for Sitecore developers.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Where can I learn more about Sitecore Headless SDKs?
+* [Sitecore JavaScript Rendering SDK](https://jss.sitecore.com/)
+* [Sitecore ASP.NET Core Rendering SDK](https://doc.sitecore.com/developers/101/developer-tools/en/sitecore-headless-development.html)
```
Comment (anastasiya29) [inline README.md]: This links to the docs on Headless Services - maybe that should be called out on it's own?
.NET Core SDK page = https://doc.sitecore.com/developers/101/developer-tools/en/sitecore-asp-net-rendering-sdk.html

File: README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+## Are these examples supported by Sitecore?
+Example code in this repository is not supported by Sitecore Product Support Services. Please do not submit support tickets regarding these examples.
+
+Individual samples are supported on GitHub by their contributors (from Sitecore or the community) but are not necessarily expected to be upgraded/updated with each release, and generally are provided as-is for you to learn from and adapt to your requirements.
```
Comment (anastasiya29) [inline README.md:12]:
```suggestion
Individual samples are supported on GitHub by their contributors (from Sitecore or the community). They are not guaranteed to be upgraded/updated with each release, and generally are provided as-is for you to learn from and adapt to your requirements. If you have trouble implementing a sample from this repository, be sure to check for differences between your version of Headless SDKs and the version used in the sample.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Contributing
+We most definitely welcome new examples!
```
Comment (anastasiya29) [inline README.md:15]: Needs emojis

File: README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+* Examples should be submitted via pull request.
+* If you submit a new example, _please be prepared to support it_. If you are tagged on a GitHub issue or PR related to your example, you are expected to respond to it.
```
Comment (anastasiya29) [inline README.md]: The "you are expected to respond to it" part sounds a bit off-putting to me. I agree with your point, but don't want to scare people off.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+* New examples should include use of Sitecore JavaScript or ASP.NET Core Rendering SDKs, or other headless consumption of Sitecore content, and should be placed in the `examples` folder.
```
Comment (anastasiya29) [inline README.md]:
```suggestion
* New examples should demonstrate usage of Sitecore JavaScript or ASP.NET Core Rendering SDKs, or other methods of consuming Sitecore content headlessly, and should be placed in the `examples` folder.
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+* Use the [`examples\README.template.md`](examples/README.template.md) as a base for your example `README.md` and fill it out completely.
```
Comment (anastasiya29) [inline README.md]:
```suggestion
* Use the [`examples\README.template.md`](examples/README.template.md) as a base for your example's `README.md`, and fill it out completely.
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+* Be sure your PR also includes an update to this README with a link to your new example.
+* Your example should use Sitecore Containers and should require as few steps as possible to get up and running. Alternatively, JavaScript-based examples can utilize "disconnected mode" / mock Sitecore services.
```
Comment (anastasiya29) [inline README.md:22]: Do you feel strongly about this? I think it's still pretty common to devep using on prem? Or is this the norm now.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+* You should simplify and remove as much boilerplate code as you can from your example, so that developers can easily see the implementation/integration scenario you are trying to demonstrate.
+* Sitecore reserves the right to decline examples, but don't let this discourage you. Submit issues or inquire on the [#headless](https://sitecorechat.slack.com/archives/C7JT0NRQW) channel of [Sitecore Community Slack](https://sitecore.chat/) if you want to validate an idea first.
+
+## Contents and Contributors
+
+_TODO: Examples below. Should these be organized somehow? Or wait until we have enough to be worth organizing?_
```
Comment (anastasiya29) [inline README.md]: Let's just put them in alpha order to set the precedent.

**Review by anastasiya29 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ancaemcken) [reply README.md]: Change to: 
```
* [Sitecore JavaScript Rendering SDK](https://doc.sitecore.com/en/developers/hd/190/sitecore-headless-development/sitecore-javascript-rendering-sdks--jss-.html)
* [Sitecore ASP.NET Core Rendering SDK](https://doc.sitecore.com/en/developers/hd/190/sitecore-headless-development/sitecore-asp-net-rendering-sdk.html)
```

**Pull request merged**